Framework/MDAlgorithms/src/CompareMDWorkspaces.cpp

```diff
@@ -31,6 +32,153 @@ class CompareFailsException : public std::runtime_error {
 namespace Mantid {
 namespace MDAlgorithms {
 
+class SimpleMDEvent {
+
+private:
+  std::vector<float> mCoordinates;
+  float mSignal;
+  float mError;
+
+public:
+  SimpleMDEvent(const std::vector<float> &coordinates, const float &signal, const float &error)
+      : mCoordinates(coordinates), mSignal(signal), mError(error) {}
+
+  SimpleMDEvent(const SimpleMDEvent &other)
+      : mCoordinates(other.mCoordinates), mSignal(other.mSignal), mError(other.mError) {}
+
+  // pretty output
+  std::string str() const {
+    std::stringstream ss;
+    size_t numdirs = mCoordinates.size();
+    for (size_t i = 0; i < numdirs; ++i)
+      ss << mCoordinates[i] << ", ";
+    ss << "signal = " << mSignal << ", error = " << mError;
+
+    return ss.str();
+  }
+
+  float getCenter(size_t dim) const { return mCoordinates[dim]; }
+
+  float getSignal() const { return mSignal; }
+
+  float getError() const { return mError; }
+
+  bool operator()(SimpleMDEvent &event1, const SimpleMDEvent &event2) { return event1 < event2; }
```

Comment:
Why are you allowing `SimpleMDEvent` instances to be called?

Reply:
This is a requirement by std::sort.

Reply:
Can you point to me where's that requirement? I can sort a vector of `int` objects with `std::sort` and the `int` object doesn't have the `()` operator.

Reply:
it is because integer has its own `()` operator.  You don't need it for a structure either.  I can try to compile without it on MacOSX.

Reply:
> it is because integer has its own `()` operator
What? Integer variables can't be called. Can you give me a code example of an integer using operator `( )`?

Reply:
What I meant is that std::sort works on integer as if they had operator ().  Even `struct` may not need operator ().